test/test_optim.py

```diff
@@ -2707,6 +2707,21 @@ def test_sgd(self):
         state = {'momentum_buffer': mbuff}
         gradcheck(_diff_fn, (p, grad, state, torch.optim.SGD, {'lr': 0.9, 'differentiable': True}, *state.values()))
 
+    def test_adam(self):
+        state = {}
+        p = torch.rand(10, requires_grad=True, dtype=torch.float64)
+        grad = torch.rand(10, requires_grad=True, dtype=torch.float64)
+        state['step'] = torch.tensor(10., requires_grad=False, dtype=torch.float64)
```

Comment:
not requires grad?

Reply:
this causes the gradcheck to fail if its enabled ...

Reply:
https://github.com/pytorch/pytorch/blob/master/torch/optim/adam.py#L140-L141
It was defined as float in the optimizer itself

Reply:
Ok,
Can you leave a comment here that mentions that step is not a continuous variable (even though we define it as a float) and so it shouldn't require gradients.
btw can we have an nice assert in the `adam()` function to ensure that step never requires grad?